### PR TITLE
Checking in code to deploy HostedEngine over gluster

### PR DESCRIPTION
**Pull request by kasturinarra**

### CODE REVIEW

File: cases/__init__.py [hunk 1]
```diff
@@ -2,4 +2,4 @@
 
 __all__ = ['CONF']
 
-CONF = list(yaml.load_all(open("./config.yml")))[0]
+CONF = list(yaml.load_all(open("/home/ramakasturinarra/automation/cockpit-auto/config.yml")))[0]
```
Comment (dracher) [inline cases/__init__.py]: don't hard code local path into constants, consider make it dynamic or put it in separate config file

File: cases/v41/const.py [hunk 2]
```diff
@@ -17,22 +16,10 @@
                 18382, 18383, 18384, 18377, 18385, 
                 18386, 18387, 18390]
 
+common_tools = ['check_subscription_rhsm', 'check_subscription_key', 'check_subscription_password']
+common_tools_id = [18412, 18413, 18414]
```
Comment (AlertBear) [inline cases/v41/const.py:20]: for test scenario reason, tools cases are splited into two parts: common_tools are used by test_common_tools.py, while common_tools_subscription are used by test_common_tools_subscription.py. So here should not merge them and remove 18410, 18416 cases.

File: cases/v41/const.py [hunk 3]
```diff
@@ -41,91 +28,53 @@
...
-dashboard_ui_id = [18534, 18535, 18536, 18540, 18541, 18542, 18543]
-
-
-# test_dashboard_ui_efi
-dashboard_ui_efi = [
-    'check_nodestatus_efi'
+    'check_ssh_key_func', 'check_nodestatus_efi', 'check_nodestatus_fc'
 ]
-dashboard_ui_efi_id = [18539]
+dashboard_ui_id = [18534, 18535, 18536, 18540, 18541, 18542, 18543, 18539, 18538]
```
Comment (AlertBear) [inline cases/v41/const.py:36]: dashboard_ui_efi and dashboard_ui_fc are just for efi machine and fc machine test, It's better to leave them alone.

File: cases/v41/const.py [hunk 4]
```diff
@@ -41,91 +28,53 @@
...
 
 
-# test_dashboard_ui_fc
-dashboard_ui_fc = [
-    'check_nodestatus_fc'
+he_install = [
+    'check_he_install', 'check_he_install_bond', 
+    'check_he_install_vlan', 'check_he_install_bv'
```
Comment (AlertBear) [inline cases/v41/const.py:41]: check_he_install_bond, check_he_install_vlan, check_he_install_bv are target for installing HE over a bond/vlan/bond+vlan network device, so they are mutual exclusion in each test. Please just keep them as original.

File: cases/v41/const.py [hunk 5]
```diff
@@ -41,91 +28,53 @@
...
+    'check_cleanup_of_gluster_setup_done',
+    'check_deployment_with_hostedengine_on_gluster',
+    'check_gluster_deployment_wizard',
+    'validate_host_deployment_tab',
+    'check_back_and_cancel_buttons_on_gdeploy_wizard',
+    'validate_arbiter_volume_creation',
+    'validate_packages_tab'
+]
+
+deployment_cases_RHHI_id =[138, 118, 119, 121, 122, 127, 129, 130, 132, 144, 145, 148, 147, 146]
```
Comment (AlertBear) [inline cases/v41/const.py]: I won't add comment one by one, we would suggest to keep existing cases and id as original, and just add deployment_ cases_RHHI and deployment_cases_RHHI_id is much better.

File: config.yml [hunk 6]
```diff
@@ -2,36 +2,52 @@
...
-    rhn_password: VHVFhPS5TEG8dud9
-    activation_key: rhevh
-    activation_org: "711497"
-    satellite_ip: "10.73.75.177"
-    satellite_hostname: satellite62.lab.eng.pek2.redhat.com
-    satellite_user: admin
-    satellite_password: redhat
-    ca_path: https://10.73.75.177/pub/katello-ca-consumer-satellite62.lab.eng.pek2.redhat.com-1.0-1.noarch.rpm
+    rhn_user: knarra_grafton_pod
+    rhn_password: graftonqa
```
Comment (AlertBear) [inline config.yml]: Even we did not notice that this is senstive data for we internal, could you please leave it blank.

File: pages/__init__.py [hunk 7]
```diff
@@ -2,4 +2,4 @@
 
 __all__ = ['CONF']
 
-CONF = list(yaml.load_all(open("./config.yml")))[0]
+CONF = list(yaml.load_all(open("/home/ramakasturinarra/automation/cockpit-auto/config.yml")))[0]
```
Comment (AlertBear) [inline pages/__init__.py]: don't hard code local path into constants

File: utils/helpers.py [hunk 8]
```diff
@@ -104,10 +104,19 @@ def wrapper(*args, **kw):
...
+                func(*args, **kw)
+            except Exception as e:
+                log.info('func(%s)|| {"RHHI-%d": "failed"}' % (func.__name__, check_id[func.__name__]))
+                log.error(e)
+            else:
+                log.info('func(%s)|| {"RHHI-%d": "passed"}' % (func.__name__, check_id[func.__name__]))
+            finally:
+                log.info('Finished to run test cases:["RHHI-%d"]' % check_id[func.__name__])
+        return wrapper
+    return decorator
```
Comment (AlertBear) [inline utils/helpers.py]: Before we did not consider any other cases id such as "RHHI-137", so we just hard code "RHEVM:" in checkpoint function. The best solution is to change id format as "RHEVM-18317" or "RHHI-137", so you do not need to add another checkpoint1 decorator.
That means:
```
log.info('func(%s)|| {"%s": "failed"}' % (func.__name__, check_id[func.__name__]))
check_id[func.__name__] will be "RHHI-137" or "RHEVM-18317".
```
What's your opinion? If yes, could you please help to change the id format in const.py. Thanks.

File: cases/v41/const.py (hunk 5 above)
Comment (kasturinarra) [reply cases/v41/const.py]: I am sorry, i did not quite understand what you meant here. Can you please elaborate?

File: config.yml (hunk 6 above)
Comment (kasturinarra) [reply config.yml]: sure, i can do that

File: pages/__init__.py (hunk 7 above)
Comment (kasturinarra) [reply pages/__init__.py]: changed it as before

File: utils/helpers.py (hunk 8 above)
Comment (kasturinarra) [reply utils/helpers.py]: yup, sure. Works for me. Will modify accordingly

File: cases/v41/const.py (hunk 2 above)
Comment (kasturinarra) [reply cases/v41/const.py:20]: These changes are not from me. Not sure how they appeared here

File: cases/v41/const.py (hunk 3 above)
Comment (kasturinarra) [reply cases/v41/const.py:36]: changes are not from me. Not sure how they appeared here

File: cases/v41/const.py (hunk 4 above)
Comment (kasturinarra) [reply cases/v41/const.py:41]: Not sure about these changes as well

**Comment by kasturinarra** [comment]
I have addressed the comments and again pushed the code. Can you please review ?

File: cases/v41/const.py (hunk 2 above)
Comment (AlertBear) [reply cases/v41/const.py:20]: That's much weird. I have to revert this file after merge your commits.

**Comment by AlertBear** [comment]
@kasturinarra ,  thanks for the last 3 comments, LGTM

**Pull request merged**